sys/net/application_layer/dns/dns.c

```diff
@@ -162,7 +162,7 @@ static int _parse_dns_reply(uint8_t *buf, size_t len, void* addr_out, int family
         }
 
         memcpy(addr_out, bufpos, addrlen);
-        return addrlen;
+        return 0;
```

Comment:
In l.226, the return value of is `_parse_dns_reply()` checked if it is >0. I'm also not sure which doc you are referring to in your commit message... `_parse_dns_reply()` does not have any.

Reply:
The documentation of [`sock_dns_query()`](https://doc.riot-os.org/group__net__sock__dns.html#ga508b47ec287a3846ca8f8fc2c0bbe9bd).

Maybe we should instead change the documentation to match the implementation - otherwise it's not possible to find out if the function returned an IPv4 or IPv6 address in the `AF_UNSPEC` case.

Reply:
But looks like the`AF_UNSPEC` case is broken too 🤷‍♂️

Reply:
But... this isn't `sock_dns_query()` but `_parse_dns_reply()`.

Reply:
The return value was simply passed on.

Reply:
Then you fix either

1. the check below as well
2. not pass on the value

;-) just changing the return value seems wrong.

Reply:
I already changed it in https://github.com/RIOT-OS/RIOT/pull/13450/commits/1c0d75f7e57a0504b5262fac50813177d310bbef, I'm just not happy with the API.